Packages/com.unity.inputsystem/InputSystem/Editor/AssetImporter/InputActionImporter.cs

```diff
@@ -67,6 +68,19 @@ private static InputActionAsset CreateFromJson(AssetImportContext context)
                 // Attempt to parse JSON
                 asset.LoadFromJson(content);
 
+                // If this JSON was authored before we switched to enum-by-value, migrate it now
+                if (asset.m_Version < kVersion)
+                {
+                    MigrateAllEnumParams(asset);
+                    asset.m_Version = kVersion;
+
+                    if (!EditorHelpers.WriteAsset(context.assetPath, asset.ToJson()))
+                        context.LogImportError($"Could not write migrated JSON to '{context.assetPath}'");
+
+                    EditorUtility.SetDirty(asset);
```

Comment:
If you really need to do this, wouldn't it be possible to use an internal overloaded LoadJson returning e.g. "isDirty" flag from the load to indicate it should be marked dirty by editor code?